packages/patternfly-4/react-core/src/components/Switch/Switch.tsx

```diff
@@ -1,49 +1,52 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import { HTMLProps, FormEvent } from 'react';
 import styles from '@patternfly/react-styles/css/components/Switch/switch';
 import { css } from '@patternfly/react-styles';
 import { CheckIcon } from '@patternfly/react-icons';
 import { getUniqueId } from '../../helpers/util';
+import { Omit } from '../../helpers/typeUtils';
 
-const propTypes = {
+export interface SwitchProps extends Omit<HTMLProps<HTMLInputElement>, 'type' | 'onChange' | 'disabled' | 'label'> {
   /** id for the label. */
-  id: PropTypes.string,
+  id?: string,
   /** Additional classes added to the Switch */
-  className: PropTypes.string,
+  className?: string,
   /** Text value for the label */
-  label: PropTypes.string,
+  label?: string,
   /** Flag to show if the Switch is checked. */
-  isChecked: PropTypes.bool,
+  isChecked?: boolean,
   /** Flag to show if the Switch is disabled. */
-  isDisabled: PropTypes.bool,
+  isDisabled?: boolean,
   /** A callback for when the Switch selection changes. (isChecked, event) => {} */
-  onChange: PropTypes.func,
+  onChange?(checked: boolean, event: FormEvent<HTMLInputElement>): void;
   /** Adds accessible text to the Switch, and should describe the isChecked="true" state. When label is defined, aria-label should be set to the text string that is visible when isChecked is true. */
-  'aria-label': props => {
-    if (!props.id && !props['aria-label']) {
-      return new Error('Switch requires either an id or aria-label to be specified');
-    }
-    return null;
-  },
-  /** Additional props are spread to the container <input> */
-  '': PropTypes.any // eslint-disable-line react/require-default-props
+  'aria-label'?: string
 };
 
-const defaultProps = {
-  id: '',
-  className: '',
-  label: '',
-  isChecked: true,
-  isDisabled: false,
-  onChange: () => undefined,
-  'aria-label': ''
-};
+export class Switch extends React.Component<SwitchProps> {
+  id = '';
```

Comment:
Remove this class attribute

Reply:
Not sure what the purpose of this id is ... is it suppose to be the same as props.id?

Reply:
it instantiates the id when it is generated. And there is no id prop set.

Reply:
Shouldn't it instantiate it as a defaultProp to `getUniqueId()`?

Reply:
I am not sure.  Does it work differently in TS than JS.  In JS if you set defaultProp to getUniqueId, if you had more  hat one Switch on the page the id's would all be the same?  That is why we had it in the constructor before.

Reply:
Looks like it will set it for all of the name to be the same based off the time.  Leave it in the constructor.